### PR TITLE
Storage v0 prep work

### PR DESCRIPTION
**Pull request by reichert621**
This PR adds documentation and addresses the feedback around Storage in preparation for beta testing.

Updates:
- Start setting up storage documentation
- Don't implicitly call getDownloadUrl after upload
- Format file size more nicely
- Allow collaborators to view/delete files in dashboard
- Add support for bulk deleting objects in dashboard storage tab
- Include storage usage in billing

### CODE REVIEW

File: client/www/components/dash/Storage.tsx [hunk 1]
```diff
@@ -612,3 +661,65 @@ export function StorageTab({
...
+          rel="noopener noreferer"
+        >
+          <Button className="mt-4" size="large">
+            Request beta access
+          </Button>
+        </a>
+      </div>
+    </div>
+  );
+}
```
Comment (reichert621) [inline client/www/components/dash/Storage.tsx:677]: Screenshot 2024-08-26 at 10 47 30 AM

File: server/src/instant/dash/routes.clj [hunk 2]
```diff
@@ -1209,6 +1219,7 @@
 
   (POST "/dash/apps/:app_id/rename" [] app-rename-post)
 
+  (GET "/dash/apps/:app_id/storage/enabled" [] storage-enabled-get)
```
Comment (stopachka) [inline server/src/instant/dash/routes.clj]: nit: perhaps we should include this in the `dash-get` route? 

This way, there would be no loading screen when the user clicks 'storage'.

File: server/src/instant/storage/beta.clj [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          get-attr-id (fn [prop]
+                        (:id (attr-model/seek-by-fwd-ident-name [whitelist-ns prop]
+                                                                attrs)))
+          patterns [[:ea '?e (get-attr-id "appId") app-id]
+                    [:ea '?e (get-attr-id "isEnabled") true]]
+          query-result (d/query {:app-id config-app-id
+                                 :db {:conn-pool aurora/conn-pool}}
+                                patterns)
+          value (get-in query-result [:symbol-values '?e])]
+      (not (empty? value)))
```
Comment (dwwoelfel) [inline server/src/instant/storage/beta.clj]: I would just double check that `(contains? value app-id)` is true here.

**Review by stopachka — APPROVED** [review]
SGTM!

File: client/www/components/dash/Storage.tsx (hunk 1 above)
Comment (stopachka) [reply client/www/components/dash/Storage.tsx:677]: Would wordsmith this. 

Maybe: 

We're working on making storage just right, and can't wait to share it with you. Are you interested in trying it out early? You can click here to request beta access. 

^ Maybe this is too obsequious, but the main goal was to say: 'we can't wait to give it to you'.

File: client/www/components/dash/Storage.tsx [hunk 4]
```diff
@@ -419,3 +661,65 @@ export function StorageTab({
...
+  className?: string;
+  app: InstantApp;
+}) {
+  return (
+    <div className={cn('flex-1 flex flex-col', className)}>
+      <div className="flex-1 flex flex-col items-center justify-center">
+        <div className="text-center text-lg font-medium text-gray-900">
+          Storage is not enabled for this app yet!
+        </div>
+        <div className="text-center text-base text-gray-500">
```
Comment (stopachka) [inline client/www/components/dash/Storage.tsx]: nit: would add a bit more spacing, and use `h2` / `p` tag.

File: server/src/instant/dash/routes.clj (hunk 2 above)
Comment (reichert621) [reply server/src/instant/dash/routes.clj]: good idea!

File: client/www/components/dash/Storage.tsx (hunk 1 above)
Comment (reichert621) [reply client/www/components/dash/Storage.tsx:677]: I like it!

**Comment by reichert621** [comment]
@stopachka wanna take one last look?

added whitelisting for beta access:
- Storage tab shows up in dashboard now
- If app is not whitelisted, displays message to request access
- Users can request via Google Form, at which point we can add them to the whitelist

(we'll also want to bump the npm package version after merging)

File: client/www/components/dash/Storage.tsx [hunk 5]
```diff
@@ -30,8 +37,27 @@ type StorageDirectory = {
   name: string;
   size: number;
   lastModified: number;
+  files: StorageFile[];
 };
 
+async function fetchStorageEnabled(
```
Comment (stopachka) [inline client/www/components/dash/Storage.tsx]: nit: we can delete this?

**Review by stopachka — APPROVED** [review]
LGTM!

**Pull request merged**